frontend/model/captureLogs.js

```diff
@@ -0,0 +1,186 @@
+    markers.push(lastEntry)
+    localStorage.setItem('giConsole/markers', JSON.stringify(markers))
+  }
+
+  if (entriesCount >= ENTRIES_LIMIT) {
+    const markers = JSON.parse(localStorage.getItem('giConsole/markers')) || []
+    let toDelete = ENTRIES_MARKER // delete the latest x entries
+    let prevEntry = markers.shift()
+    do {
+      const entry = JSON.parse(localStorage.getItem(`giConsole/${prevEntry}`)) || {}
+      localStorage.removeItem(`giConsole/${prevEntry}`)
+      prevEntry = entry.prev
+      entriesCount--
+      toDelete--
+    } while (toDelete && prevEntry)
+    localStorage.setItem('giConsole/markers', JSON.stringify(markers))
+    localStorage.setItem('giConsole/count', entriesCount)
+  }
+}
+
+export function captureLogsStart () {
+  isCapturing = true
+  lastEntry = localStorage.getItem('giConsole/lastEntry')
+  entriesCount = +localStorage.getItem('giConsole/count') || 0
+
+  // Set a new visit or session - useful to understand logs through time.
+  // NEW_SESSION -> The user opened a new browser or tab.
+  // NEW_VISIT -> The user comes from an ongoing session (refresh or login)
+  const isNewSession = !sessionStorage.getItem('NEW_SESSION')
+  if (isNewSession) { sessionStorage.setItem('NEW_SESSION', 1) }
+  captureLog(isNewSession ? 'NEW_SESSION' : 'NEW_VISIT', Date.now())
+
+  // Subscribe to appLogs applied filter
+  const state = sbp('state/vuex/state')
+  const userContract = state[state.loggedIn.identityContractID] || {}
+  appLogsFilter = (userContract.settings || {}).appLogsFilter || []
+  sbp('okTurtles.events/on', SET_APP_LOGS_FILTER, filter => { appLogsFilter = filter })
+
+  console.debug('Starting to capture logs of type:', appLogsFilter)
+
+  // Override the console to start capturing the logs
+  if (!isConsoleOveride) {
+    // avoid duplicated captures, in case captureLogsStart
+    // is called multiple times. (ex: login twice in the same visit)
+    isConsoleOveride = true
+
+    const _cError = console.error
+    const _cWarn = console.warn
+    const _cDebug = console.debug
+    const guardCapture = (type, args) => {
+      if (isCapturing && appLogsFilter.includes(type)) {
+        captureLog(type, ...args)
+      }
+    }
+
+    console.error = function () {
+      _cError.apply(console, arguments)
+      guardCapture('error', arguments)
+    }
+    console.warn = function () {
+      _cWarn.apply(console, arguments)
+      guardCapture('warn', arguments)
+    }
+    console.debug = function () {
+      _cDebug.apply(console, arguments)
+      guardCapture('debug', arguments)
+    }
+  }
+
+  // // REMOVE THIS: It overrides all types,
+  // // (including log and info). It's not needed.
+  // window.console = new Proxy(window.console, {
+  //   get: (obj, type) => (...args) => {
+  //     obj[type](...args)
+  //     if (isCapturing && appLogsFilter.includes(type)) {
+  //       captureLog(type, ...args)
+  //     }
+  //   }
+  // })
```

Comment:
So, should I implement the checkbox for `log` and `info` too on the UI but unchecked? The way this is done only the filtered logs are sent to us (because only those are stored on LS)

Reply:
Sure, but leave `info` checked by default.  We might also want to change `console.log` to `console.debug` in `main.js:53` and elsewhere later, but for now we can just leave them out of the logs by leaving `log` unchecked for now.